plumbum.py

```diff
@@ -78,37 +79,26 @@ def lookup(instances, filter_by=None):
     return instances
 
 
-def interpret_options(options):
-    """Parse all the command line options."""
-    # template always has to be index 0
-    template = options[0]
-    # namespace always has to be index 1. Support 'ec2' (human friendly) and
-    # 'AWS/EC2' (how CloudWatch natively calls these things)
-    namespace = options[1].rsplit('/', 2)[-1].lower()
-    next_idx = 2
-    # region might be index 2
-    region = ''
-    if len(options) > 2 and re.match(r'^\w+\-[\w\-]+\-\d+$', options[2]):
-        region = options[2]
-        next_idx += 1
-    else:
-        next_idx = 2
-    region = region or boto.config.get('Boto', 'ec2_region_name', 'us-east-1')
-
-    filter_by = {}
-    extras = []
-    for arg in options[next_idx:]:
-        if arg.startswith('-'):
-            # throw these away for now
-            extras.append(arg)
-        elif '=' in arg:
-            key, value = arg.split('=', 2)
-            filter_by[key] = value
-        else:
-            # throw these away for now
-            extras.append(arg)
-
-    return template, namespace, region, filter_by, extras
+def interpret_options(args=sys.argv[1:]):
+
+    if '--version' in args:
+        print(__version__)
+        sys.exit()
```

Comment:
Generally `sys.exit` from inside a function is no bueno, but given the rest of the code it seems fine here. Just keep in mind it's usually an anti-pattern.